signer/src/request_decider.rs

```diff
@@ -345,12 +345,23 @@ where
             return Ok(true);
         };
 
-        let result = client
-            .can_accept(&req.sender_address.to_string())
+        let receiver_pubkey = req.recipient.clone();
```

Comment:
You don't need to clone this, you can just take a reference to `req.recipient` below.